starknet/test/stark-sig-auth.test.ts

```diff
@@ -0,0 +1,256 @@
+  let domain: any;
+
+  beforeAll(async () => {
+    // Deploy Starknet Signature Authenticator
+    const starkSigAuthSierra = json.parse(
+      fs.readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.sierra.json').toString('ascii'),
+    );
+    const starkSigAuthCasm = json.parse(
+      fs.readFileSync('starknet/target/dev/sx_StarkSigAuthenticator.casm.json').toString('ascii'),
+    );
+
+    let deployResponse = await account0.declareAndDeploy({
+      contract: starkSigAuthSierra,
+      casm: starkSigAuthCasm,
+      constructorCalldata: CallData.compile({ name: 'sx-sn', version: '0.1.0' }),
+    });
+    starkSigAuthAddress = deployResponse.deploy.contract_address;
+
+    // Deploy Vanilla Voting Strategy
+    const vanillaVotingStrategySierra = json.parse(
+      fs.readFileSync('starknet/target/dev/sx_VanillaVotingStrategy.sierra.json').toString('ascii'),
+    );
+    const vanillaVotingStrategyCasm = json.parse(
+      fs.readFileSync('starknet/target/dev/sx_VanillaVotingStrategy.casm.json').toString('ascii'),
+    );
+
+    deployResponse = await account0.declareAndDeploy({
+      contract: vanillaVotingStrategySierra,
+      casm: vanillaVotingStrategyCasm,
+      constructorCalldata: CallData.compile({}),
+    });
+    vanillaVotingStrategyAddress = deployResponse.deploy.contract_address;
+
+    // Deploy Vanilla Proposal Validation Strategy
+    const vanillaProposalValidationStrategySierra = json.parse(
+      fs
+        .readFileSync('starknet/target/dev/sx_VanillaProposalValidationStrategy.sierra.json')
+        .toString('ascii'),
+    );
+    const vanillaProposalValidationStrategyCasm = json.parse(
+      fs
+        .readFileSync('starknet/target/dev/sx_VanillaProposalValidationStrategy.casm.json')
+        .toString('ascii'),
+    );
+
+    deployResponse = await account0.declareAndDeploy({
+      contract: vanillaProposalValidationStrategySierra,
+      casm: vanillaProposalValidationStrategyCasm,
+      constructorCalldata: CallData.compile({}),
+    });
+    vanillaProposalValidationStrategyAddress = deployResponse.deploy.contract_address;
+
+    // Deploy Space
+    const spaceSierra = json.parse(
+      fs.readFileSync('starknet/target/dev/sx_Space.sierra.json').toString('ascii'),
+    );
+    const spaceCasm = json.parse(
+      fs.readFileSync('starknet/target/dev/sx_Space.casm.json').toString('ascii'),
+    );
+
+    deployResponse = await account0.declareAndDeploy({
+      contract: spaceSierra,
+      casm: spaceCasm,
+      constructorCalldata: CallData.compile({
+        _owner: 1,
+        _max_voting_duration: 100,
+        _min_voting_duration: 100,
+        _voting_delay: 15,
+        _proposal_validation_strategy: {
+          address: vanillaProposalValidationStrategyAddress,
+          params: [[]],
+        },
+        _proposal_validation_strategy_metadata_URI: [],
+        _voting_strategies: [{ address: vanillaVotingStrategyAddress, params: [] }],
+        _voting_strategies_metadata_URI: [],
```

Comment:
needs to be `[[]]` now that space checks whether `_voting_strategies.len() == _voting_strategies_metadata_uri.len()`

also, all occurences of `URI` should be replaced by `uri` or `Uri` (at least for stuff interacting with cairo contracts)